generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -0,0 +1,180 @@
+package com.scottlogic.deg.generator.constraints;
+
+import com.scottlogic.deg.generator.constraints.atomic.*;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StringConstraintsCollection {
+    private final Set<AtomicConstraint> constraints;
+
+    public StringConstraintsCollection(Set<AtomicConstraint> constraints) {
+        this.constraints = constraints;
+    }
+
+    public StringConstraintsCollection(AtomicConstraint constraint){
+        this(Collections.singleton(constraint));
+    }
+
+    public StringConstraintsCollection intersect(StringConstraintsCollection otherConstraint){
```

Comment:
This type is trying to reflect data, with the ability to operate on the data.
As such it can then be used in the merger rather than being the merger itself.

Reply:
Renamed to union as this is what it actually does.